test/suite/test_compress03.py

```diff
@@ -0,0 +1,113 @@
+
+import wttest
+from wiredtiger import stat
+from wtdataset import SimpleDataSet
+from wtscenario import make_scenarios
+
+# test_compress03.py
+#   Verify the statistics related to compression are updated as expected.
+#
+class test_compress03(wttest.WiredTigerTestCase):
+
+    compressors = [
+        ('lz4', dict(compressors='lz4')),
+        ('snappy', dict(compressors='snappy')),
+        ('zlib', dict(compressors='zlib')),
+        ('zstd', dict(compressors='zstd')),
+    ]
+    scenarios = make_scenarios(compressors)
+
+    uri = "table:test_compress03"
+    nrows = 100000
+    valuea = "aaaaa"
+    valueb = "bbbbb"
+
+    def conn_config(self):
+        return f'statistics=(fast),statistics_log=(json,on_close,wait=1,sources=(\"file:\"))'
+
+    # Load the compression extension, skip the test if missing.
+    def conn_extensions(self, extlist):
+        extlist.skip_if_missing = True
+        extlist.extension('compressors', self.compressors)
+
+    def verify_stats(self, uri):
+        stat_cursor = self.session.open_cursor('statistics:' + uri, None, 'statistics=(fast)')
+        total_reads = stat_cursor[stat.dsrc.compress_read][2]
+        total_reads_hist = stat_cursor[stat.dsrc.compress_read_hist_ratio_2][2]
+        total_reads_hist += stat_cursor[stat.dsrc.compress_read_hist_ratio_4][2]
+        total_reads_hist += stat_cursor[stat.dsrc.compress_read_hist_ratio_8][2]
+        total_reads_hist += stat_cursor[stat.dsrc.compress_read_hist_ratio_16][2]
+        total_reads_hist += stat_cursor[stat.dsrc.compress_read_hist_ratio_32][2]
+        total_reads_hist += stat_cursor[stat.dsrc.compress_read_hist_ratio_64][2]
+        total_reads_hist += stat_cursor[stat.dsrc.compress_read_hist_ratio_max][2]
+        # We expect at least a read of compressed data.
+        assert total_reads and total_reads == total_reads_hist
+
+        total_writes = stat_cursor[stat.dsrc.compress_write][2]
+        total_writes_hist = stat_cursor[stat.dsrc.compress_write_hist_ratio_2][2]
+        total_writes_hist += stat_cursor[stat.dsrc.compress_write_hist_ratio_4][2]
+        total_writes_hist += stat_cursor[stat.dsrc.compress_write_hist_ratio_8][2]
+        total_writes_hist += stat_cursor[stat.dsrc.compress_write_hist_ratio_16][2]
+        total_writes_hist += stat_cursor[stat.dsrc.compress_write_hist_ratio_32][2]
+        total_writes_hist += stat_cursor[stat.dsrc.compress_write_hist_ratio_64][2]
+        total_writes_hist += stat_cursor[stat.dsrc.compress_write_hist_ratio_max][2]
+        # We expect at least a write of compressed data.
+        assert total_writes and total_writes == total_writes_hist
+
+        stat_cursor.close()
+
+    def large_updates(self, uri, value, ds, nrows, ts):
+        cursor = self.session.open_cursor(uri)
+        for i in range(0, nrows):
+            self.session.begin_transaction()
+            cursor[ds.key(i)] = value
+            self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(ts))
```

Comment:
This test probably doesn't need the explicit transaction and timestamp assignment.

Reply:
That is true, removed in e3b2546034aad674d5421a6b6a61466118b2338e.